python/python_direct/ops.cpp

```diff
@@ -13,9 +13,39 @@ namespace nvfuser::python {
 
 namespace {
 
+// ScalarVariant represents a NvFuser Val or python scalar.
+using ScalarVariant = std::variant<Val*, PolymorphicValue::VariantType>;
+
+// This function converts a ScalarVariant to a nvfuser Val.
+Val* convertToVal(
+    ScalarVariant value,
+    std::optional<PrimDataType> dtype = std::nullopt) {
+  // short-circuit: already a NvFuser val
+  if (std::holds_alternative<Val*>(value)) {
+    return std::get<Val*>(value);
```

Comment:
I think we should either assert dtype is std::nullopt, alternatively, we should have a ternary op and try to cast like how we are handling python values.